provider/core/src/buffer_provider.rs

```diff
@@ -0,0 +1,18 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Traits for data providers that produce opaque buffers.
+
+use crate::error::Error;
+use crate::prelude::*;
+
+pub struct BufferMarker {}
```

Comment:
nit: standard rust style for unit structs

```suggestion
pub struct BufferMarker;
```